remote/bitbucketserver/internal/client.go

```diff
@@ -214,3 +250,142 @@ func (c *Client) paginatedRepos(start int) ([]*Repo, error) {
 	}
 	return repoResponse.Values, nil
 }
+
+func Filter(vs []string, f func(string) bool) []string {
+	vsf := make([]string, 0)
+	for _, v := range vs {
+		if f(v) {
+			vsf = append(vsf, v)
+		}
+	}
+	return vsf
+}
+
+//TODO: find a clean way of doing these next two methods- bitbucket server hooks only support 20 cb hooks
+func arrayToHookSettings(hooks []string) HookSettings {
```

Comment:
yeah, that's the only way I could think of doing it otherwise the readability would take a hit which it was I did it this way, until I could think of a "simpler" way.  Do you need me to remove the comments?